clang/lib/CIR/Dialect/IR/CIRAttrs.cpp

```diff
@@ -214,6 +214,23 @@ void LangAttr::print(AsmPrinter &printer) const {
   printer << "<" << getLang().getValue() << '>';
 }
 
+//===----------------------------------------------------------------------===//
+// OptLevelAttr definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult OptLevelAttr::verify(function_ref<InFlightDiagnostic()> emitError,
+                                   unsigned optLevel, unsigned optSizeLevel) {
+  if (optLevel > 3) {
+    emitError() << "invalid optimization level";
```

Comment:
I'd prefer the error explicitly indicating the valid range.

```suggestion
    emitError() << "optimization level must be between 0 and 3";
```

Reply:
Updated.